spec/models/ability_spec.rb

```diff
@@ -1,55 +1,140 @@
+    Notice.new => { yes: [:display, :image], no: standard },
+    PageElement.new => { yes: [], no: standard },
+    Page.new => { yes: [:read], no: [:create, :update, :destroy] },
+    Permission.new => { yes: [], no: standard },
+    Post.new => { yes: [:collapse, :display, :show], no: [:index, :create, :update] },
+    Rent.new => { yes: [:main, :create, :index], no: [:update, :destroy] }
+  }
+
+  subject(:not_signed_ability) { Ability.new(nil) }
+
+  let(:signed) { create(:user, member_at: nil) }
+  subject(:signed_ability) { Ability.new(signed) }
+
+  let(:member) { create(:user, member_at: Time.zone.now) }
+  subject(:member_ability) { Ability.new(member) }
+
+  describe 'Not signed in' do
+    unsigned.each do |obj, value|
+      if value[:yes].present?
+        it { not_signed_ability.should have_abilities(value[:yes], obj) }
+      end
+
+      if value[:no].present?
+        it { not_signed_ability.should not_have_abilities(value[:no], obj) }
+      end
+    end
   end
 
-  context 'default user' do
-    let(:user) { create(:user) }
-    let(:ability) { create(:user) }
-    subject(:ability) { Ability.new(user) }
-
-    # Stuff everyone can do
-    it { should be_able_to :read, Event.new }
-    it { should be_able_to :read, Council.new }
-    #it { should be_able_to :read, CafeWork.new }
-    it { should be_able_to :read, News.new }
-    it { should be_able_to :read, Post.new }
-    it { should be_able_to :read, Election.new }
-
-
-    # Stuff everyone must not be able to do
-    it { should_not be_able_to :read, Constant.new }
-    it { should_not be_able_to :read, Album.new }
-    it { should_not be_able_to :read, Candidate.new }
-    it { should_not be_able_to :read, Permission.new }
-    it { should_not be_able_to :read, Profile.new }
-    it { should_not be_able_to :read, User.new }
+  describe 'Signed in' do
+    ab_signed.each do |obj, value|
+      if value[:yes].present?
+        it { signed_ability.should have_abilities(value[:yes], obj) }
+      end
+
+      if value[:no].present?
+        it { signed_ability.should not_have_abilities(value[:no], obj) }
+      end
+    end
+
+    # Extra cases which cannot be covered in loop
+    # These also count for the members
+    it { signed_ability.should have_abilities([:show, :update, :destroy], Rent.new(user: signed)) }
+    it { signed_ability.should
+         have_abilities([:update_worker, :remove_worker, :edit], CafeWork.new(user: signed)) }
   end
 
-  context 'admin user' do
-    let(:user) { create(:admin) }
+  describe 'Member' do
+    ab_member.each do |obj, value|
+      if value[:yes].present?
+        it { member_ability.should have_abilities(value[:yes], obj) }
+      end
+
+      if value[:no].present?
+        it { member_ability.should not_have_abilities(value[:no], obj) }
+      end
+    end
 
-    # Can do anything
-    it { should be_able_to :manage, :all }
+    # Extra cases which cannot be covered in loop
+    it { member_ability.should
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.